src/stats.bif

```diff
@@ -39,28 +39,24 @@ zeek::RecordTypePtr ReporterStats;
 ##              get_reporter_stats
 function get_net_stats%(%): NetStats
 	%{
-	uint64_t recv = 0;
-	uint64_t drop = 0;
-	uint64_t link = 0;
-	uint64_t bytes_recv = 0;
+	struct zeek::iosource::PktSrc::Stats stat;
 
 	if ( zeek::iosource::PktSrc* ps = zeek::iosource_mgr->GetPktSrc() )
 		{
-		struct zeek::iosource::PktSrc::Stats stat;
 		ps->Statistics(&stat);
-		recv += stat.received;
-		drop += stat.dropped;
-		link += stat.link;
-		bytes_recv += stat.bytes_received;
+		ps->PostStatsTelemetry(stat);
```

Comment:
> My take (currently) is that we either use Inc() directly on metrics during processing in C++. 

Part of the problem is that `PktSrc::Statistics()` is a pure-virtual method that has to be implemented by every packet source, and as such we can't count on maintainers to always update the telemetry. For the pcap source, for example, we don't have the data on packet drop counts unless Statistics() is called because that's where `pcap_stats` is called. We can obviously update `received` and `bytes_received` based on the packet data in other places, but now we're doing it in multiple locations (and expecting packet source maintainers to know to do the same). Doing it from a single location takes it out of the maintainers hands of having to deal with.

Reply:
> The following is what's used in zeek-demo to produce 3 of the 5 metrics listed here using the Telemetry::sync hook that runs every (for Prometheus a more reasonable) 10 seconds by default

Why is this part of some demo code and not actually in the zeek code base? That seems rather useful.

Reply:
Ok, looking into it more that's definitely less effort. I was unaware that all of that sample code existed for the telemetry stuff. Odd that it fit perfectly for the github issue.

Reply:
> Odd that it fit perfectly for the github issue.

Well, there have been various versions of these metrics in various places, so you'd think at some point they work out mostly :-)